sqlglot/parser.py

```diff
@@ -572,93 +539,145 @@ def parse_option(option, token, option_lambda):
+                this=exp.Literal.string(c.PARTITIONED_BY),
+                value=self._parse_schema(),
+            )
+        if self._match(TokenType.STORED):
+            self._match(TokenType.ALIAS)
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.FileFormatProperty,
+                this=exp.Literal.string(c.FILE_FORMAT),
+                value=exp.Literal.string(self._parse_var().text("this")),
+            )
+        if self._match(TokenType.LOCATION):
+            return self.expression(
+                exp.LocationProperty,
+                this=exp.Literal.string(c.LOCATION),
+                value=self._parse_string(),
+            )
+        if self._match(TokenType.FORMAT):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.FileFormatProperty,
+                this=exp.Literal.string(c.FILE_FORMAT),
+                value=self._parse_string() or self._parse_var(),
+            )
+        if self._match(TokenType.ENGINE):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.EngineProperty,
+                this=exp.Literal.string(c.ENGINE),
+                value=self._parse_var() or self._parse_string(),
+            )
+        if self._match(TokenType.AUTO_INCREMENT):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.AutoIncrementProperty,
+                this=exp.Literal.string(c.AUTO_INCREMENT),
+                value=self._parse_var() or self._parse_number(),
+            )
+        if self._match(TokenType.CHARACTER_SET) or self._match_sequence(
+            TokenType.DEFAULT, TokenType.CHARACTER_SET
+        ):
+            self._match(TokenType.EQ)
+            potential_default_token = self._get_token_at_index(self._index - 3)
+            return self.expression(
+                exp.CharacterSetProperty,
+                this=exp.Literal.string(c.CHARACTER_SET),
+                value=self._parse_var() or self._parse_string(),
+                default=potential_default_token is not None
+                and potential_default_token.token_type == TokenType.DEFAULT,
+            )
+        if self._match(TokenType.COLLATE):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.CollateProperty,
+                this=exp.Literal.string(c.COLLATE),
+                value=self._parse_var() or self._parse_string(),
+            )
+        if self._match(TokenType.SCHEMA_COMMENT):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.SchemaCommentProperty,
+                this=exp.Literal.string(c.COMMENT),
+                value=self._parse_string(),
+            )
+        if self._match_sequence(
+            TokenType.VAR, TokenType.EQ, advance=False
+        ) or self._match_sequence(TokenType.VAR, TokenType.EQ, advance=False):
```

Comment:
what's this or statement both var eq? typo?

Reply:
how come you're not advancing it? so that you can absorb the key? maybe match_sequence can return a tuple?

Reply:
Yeah typo thanks for catching.

Yeah I want to absorb the key. Personally this pattern seems fine... with later Python releases we get the walrus operator which then combined with returning the tuple could be a nice solution.

Reply:
ok